kyo-core/shared/src/main/scala/kyo/tags.scala

```diff
@@ -1,41 +1,206 @@
 package kyo
 
-import izumi.reflect.dottyreflection.TypeInspections
-import izumi.reflect.macrortti.LightTypeTag
+import scala.annotation.tailrec
 import scala.quoted.*
 
-opaque type Tag[T] = String
+case class Tag[T](tpe: String) extends AnyVal
 
 object Tag:
 
-    given canEqual[T, U]: CanEqual[Tag[T], Tag[U]] = CanEqual.derived
+    import internal.*
+
+    inline given apply[T >: Nothing]: Tag[T] = ${ tagImpl[T] }
+
+    extension [T](t1: Tag[T])
+
+        def show = t1.tpe.takeWhile(_ != ';')
```

Comment:
We should eventually improve this to show the entire type signature. This code only prints the outer class name, no type parameters. It's enough for the error reporting use case, though.

Reply:
Perhaps worth a pending test!

Reply:
Makes sense! I've just added it.